pkg/machines/scripts/create_machine.sh

```diff
@@ -44,16 +44,18 @@ if [ "$OS" = "other-os" -o  -z "$OS" ]; then
 fi
 
 FIRST_1_SOURCE="`echo "$SOURCE" | cut -c 1`"
-FIRST_3_SOURCE="`echo "$SOURCE" | cut -c -3`"
-FIRST_4_SOURCE="`echo "$SOURCE" | cut -c -4`"
 
 if [ "$START_VM" = "true" -a "$FIRST_1_SOURCE" = "/" -a ! -f "$SOURCE" ]; then
     echo "$SOURCE does not exist or is not a file" 1>&2
     exit 1
 fi
 
-if [ "$START_VM" = "true" -a \( "$FIRST_1_SOURCE" = "/" -o "$FIRST_4_SOURCE" = "http" -o "$FIRST_3_SOURCE" = "ftp" -o "$FIRST_3_SOURCE" = "nfs" \) ]; then
-    LOCATION_PARAM="--cdrom $SOURCE"
+if [ "$START_VM" = "true" ]; then
+    if [ "$FIRST_1_SOURCE" = "/" ] && [ "${SOURCE##*.}" = iso ]; then
```

Comment:
Note that the standard shell test for "starts with /" is `[ "${SOURCE#/}" != "$SOURCE"`, then you avoid the subshell and cut call entirely. You already use these shell operators in the second test :)

To make the second test a bit more steamlined, you can use `[ "${SOURCE%.iso}" != "$SOURCE" ]`. (But it should have the same effect).


Nitpick: Commit message has a "ditribution" typo.

Reply:
Done, thanks for the hints :)